front/src/applications/stdcm/components/StdcmLayersModal.tsx

```diff
@@ -0,0 +1,157 @@
+import { useCallback, useMemo, useState } from 'react';
+
+import { isString, uniq, concat, compact } from 'lodash';
+import { useTranslation } from 'react-i18next';
+import { GiElectric, GiUnplugged } from 'react-icons/gi';
+import { MdSpeed } from 'react-icons/md';
+import { TbRectangleVerticalFilled } from 'react-icons/tb';
+import { useSelector } from 'react-redux';
+
+import bufferStopIcon from 'assets/pictures/layersicons/bufferstop.svg';
+import detectorsIcon from 'assets/pictures/layersicons/detectors.svg';
+import signalsIcon from 'assets/pictures/layersicons/layer_signal.svg';
+import pslsIcon from 'assets/pictures/layersicons/layer_tivs.svg';
+import OPsSVGFile from 'assets/pictures/layersicons/ops.svg';
+import switchesIcon from 'assets/pictures/layersicons/switches.svg';
+import { osrdEditoastApi } from 'common/api/osrdEditoastApi';
+import { Modal } from 'common/BootstrapSNCF/ModalSNCF';
+import SwitchSNCF from 'common/BootstrapSNCF/SwitchSNCF/SwitchSNCF';
+import { Icon2SVG } from 'common/Map/Settings/MapSettingsLayers';
+import { useInfraID } from 'common/osrdContext';
+import type { LayersSettings } from 'reducers/map';
+import { stdcmConfInitialState, updateStdcmLayers } from 'reducers/osrdconf/stdcmConf';
+import { getStdcmSelectedLayers } from 'reducers/osrdconf/stdcmConf/selectors';
+import { useAppDispatch } from 'store';
+
+const LAYERS = [
+  { layer: 'track_sections', icon: <MdSpeed className="mx-2" /> },
+  { layer: 'signals', icon: signalsIcon },
+  { layer: 'buffer_stops', icon: bufferStopIcon },
+  { layer: 'detectors', icon: detectorsIcon },
+  { layer: 'switches', icon: switchesIcon },
+  { layer: 'sncf_psl', icon: pslsIcon },
+  { layer: 'electrifications', icon: <GiElectric className="mx-2" style={{ width: '20px' }} /> },
+  { layer: 'neutral_sections', icon: <GiUnplugged className="mx-2" style={{ width: '20px' }} /> },
+  {
+    layer: 'platforms',
+    icon: <TbRectangleVerticalFilled className="mx-2" style={{ width: '20px' }} />,
+  },
+  {
+    layer: 'operational_points',
+    icon: <Icon2SVG file={OPsSVGFile} style={{ width: '20px' }} className="mx-2" />,
+  },
+  { layer: 'speed_limits', icon: <MdSpeed style={{ width: '20px' }} className="mx-2" /> },
+];
+
+type StdcmLayersModalProps = {
+  initialLayers: LayersSettings | undefined;
```

Comment:
Make initialLayers not nullable and give stdcmConfInitialState.layersSettings by default in the parent component